src/libraries/System.Text.Json/tests/Common/ExtensionDataTests.cs

```diff
@@ -428,104 +428,93 @@ public override void Write(Utf8JsonWriter writer, CustomOverflowDictionary<JsonE
         }
 
         [Theory]
-        [InlineData(typeof(Dictionary<string, object>), typeof(DictionaryOverflowConverter))]
-        [InlineData(typeof(Dictionary<string, JsonElement>), typeof(JsonElementOverflowConverter))]
-        [InlineData(typeof(CustomOverflowDictionary<object>), typeof(CustomObjectDictionaryOverflowConverter))]
-        [InlineData(typeof(CustomOverflowDictionary<JsonElement>), typeof(CustomJsonElementDictionaryOverflowConverter))]
-        public void ExtensionProperty_SupportsWritingToCustomSerializerWithOptions(Type overflowType, Type converterType)
-        {
-            typeof(ExtensionDataTests)
-                .GetMethod(nameof(ExtensionProperty_SupportsWritingToCustomSerializerWithOptionsInternal), BindingFlags.Static | BindingFlags.NonPublic)
-                .MakeGenericMethod(overflowType, converterType)
-                .Invoke(null, null);
-        }
-
-        private static void ExtensionProperty_SupportsWritingToCustomSerializerWithOptionsInternal<TDictionary, TConverter>()
-            where TDictionary : new()
-            where TConverter : JsonConverter, new()
+        [MemberData(nameof(GetCustomOverflowConverters))]
+        public async Task ExtensionProperty_SupportsWritingToCustomSerializerWithOptions<TDictionary>(JsonConverter<TDictionary> dictionaryConverter)
+            where TDictionary : IDictionary, new()
         {
             var root = new ClassWithExtensionData<TDictionary>()
             {
                 Overflow = new TDictionary()
             };
 
-            var options = new JsonSerializerOptions();
-            options.Converters.Add(new TConverter());
+            var options = Serializer.CreateOptions(makeReadOnly: false);
+            options.Converters.Add(dictionaryConverter);
 
-            string json = JsonSerializer.Serialize(root, options);
+            string json = await Serializer.SerializeWrapper(root, options);
             Assert.Equal(@"{""MyCustomOverflowWrite"":""OverflowValueWrite""}", json);
         }
 
-        private interface IClassWithOverflow<T>
+        public interface IClassWithOverflow
```

Comment:
out of curiosity, does the visibility change have influence on trimming or are we just reusing this class elsewhere?

Reply:
It is passed as a parameter in theory method, so the type needs to be public.